src/core/crypto/ecdsa.cpp

```diff
@@ -42,6 +42,12 @@
 #include <mbedtls/pk.h>
 #include <mbedtls/version.h>
 
+#ifdef MBEDTLS_USE_TINYCRYPT
```

Comment:
I think it would be cleaner if we had two separate cpp files that represent `TINYCRYPT` disabled and enabled. While there is some shared code, it isn't much. Having two separate files would make the code more readable. Thoughts?

Reply:
I know I suggested above to see if we can reuse the common part of the code as much as possible, but looking at the src file and shared part, the code under the two cases is mostly different, so I agree with @jwhui  on this and I think it may be better to have a different cpp file one under `TINYCRYPT`. It for sure helps make the code easier to read.

Reply:
Thank you for the feedback. 
To make sure that I understand. You suggest creating another cpp file (e.g ecdsa_tinycrypt.cpp) with tinycrypt-based functions (and declare them in the ecdsa.hpp file) and then calling them in the ecdsa.cpp function based on define?

Reply:
Got it, thank you

Reply:
Should I use `#if MBEDTLS_USE_TINYCRYPT` or `#ifdef MBEDTLS_USE_TINYCRYPT`?

Reply:
I made the changes. I used #ifdef and #ifndef as this is how tinycrypt changes are integrated in the mbedtls library. Please, let me know if it's ok.